tests/diviner/test_diviner_model_export.py

```diff
@@ -0,0 +1,479 @@
+)
+
+
+DS_FORMAT = "%Y-%m-%dT%H:%M:%S"
+EXTRA_PYFUNC_SERVING_TEST_ARGS = [] if _is_available_on_pypi("diviner") else ["--no-conda"]
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return str(pathlib.Path(tmpdir).joinpath("model"))
+
+
+@pytest.fixture(scope="session")
+def diviner_data():
+    return example_data_generator.generate_example_data(
+        column_count=3, series_count=4, series_size=365 * 4, start_dt="2019-01-01", days_period=1
+    )
+
+
+@pytest.fixture(scope="session")
+def grouped_prophet(diviner_data):
+    return GroupedProphet(n_changepoints=20, uncertainty_samples=0).fit(
+        df=diviner_data.df, group_key_columns=diviner_data.key_columns, y_col="y", datetime_col="ds"
+    )
+
+
+@pytest.fixture(scope="session")
+def grouped_pmdarima(diviner_data):
+    from pmdarima.arima.auto import AutoARIMA
+
+    base_model = AutoARIMA(out_of_sample_size=60, maxiter=30)
+    return GroupedPmdarima(model_template=base_model).fit(
+        df=diviner_data.df,
+        group_key_columns=diviner_data.key_columns,
+        y_col="y",
+        datetime_col="ds",
+        silence_warnings=True,
+    )
+
+
+@pytest.fixture
+def diviner_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
+    _mlflow_conda_env(conda_env, additional_pip_deps=["diviner"])
+    return conda_env
+
+
+def test_diviner_native_save_and_load(grouped_prophet, model_path):
+
+    mlflow.diviner.save_model(diviner_model=grouped_prophet, path=model_path)
+
+    loaded = mlflow.diviner.load_model(model_path)
+
+    local_forecast = grouped_prophet.forecast(10, "D")
+    loaded_forecast = loaded.forecast(10, "D")
+
+    pd.testing.assert_frame_equal(local_forecast, loaded_forecast)
+
+
+def test_diviner_pyfunc_save_load(grouped_pmdarima, model_path):
+
+    mlflow.diviner.save_model(diviner_model=grouped_pmdarima, path=model_path)
+    loaded_pyfunc = pyfunc.load_pyfunc(model_uri=model_path)
+
+    model_predict = grouped_pmdarima.predict(n_periods=10, return_conf_int=True, alpha=0.075)
+
+    predict_conf = pd.DataFrame(
+        {"n_periods": 10, "return_conf_int": True, "alpha": 0.075}, index=[0]
+    )
+    pyfunc_predict = loaded_pyfunc.predict(predict_conf)
+
+    pd.testing.assert_frame_equal(model_predict, pyfunc_predict)
+
+
+def test_diviner_pyfunc_invalid_config_raises(grouped_prophet, model_path):
+
+    mlflow.diviner.save_model(diviner_model=grouped_prophet, path=model_path)
+    loaded_pyfunc_model = pyfunc.load_pyfunc(model_uri=model_path)
+
+    with pytest.raises(
+        MlflowException,
+        match="The provided prediction configuration Pandas "
+        "DataFrame does not contain either the `n_periods` "
+        "or `horizon` columns.",
+    ):
+        bad_conf = pd.DataFrame({"bogus": "config"}, index=[0])
+        loaded_pyfunc_model.predict(bad_conf)
+
+    with pytest.raises(
+        MlflowException,
+        match="The `n_periods` column contains invalid data. " "Supplied type must be an integer.",
```

Comment:
```suggestion
        match="The `n_periods` column contains invalid data. Supplied type must be an integer.",
```

Can we join strings?